src/components/map/map.tsx

```diff
@@ -0,0 +1,94 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Listen,
+  Prop
+} from "@stencil/core";
+import { IComponent } from "../common/interfaces";
+import {
+  Marker,
+  map as LFMap,
+  tileLayer
+  // tslint:disable-next-line:no-submodule-imports
+} from "leaflet/dist/leaflet-src.esm";
+
+@Component({
+  shadow: false,
+  styleUrl: "map.scss",
+  tag: "gx-map"
+})
+export class Map implements IComponent {
+  map: any;
+  @Element() element;
+
+  /**
+   * The initial center of the map.
+   *
+   */
+  @Prop({ mutable: true })
+  center = "0, 0";
+
+  /**
+   * The max zoom scale aviable in the map.
+   *
+   */
+  @Prop({ mutable: true })
+  maxZoom = 20;
+
+  /**
+   * The initial zoom scale in the map.
+   *
+   */
+  @Prop({ mutable: true })
+  zoom = 1;
+
+  /**
+   * Emmits when map is loaded.
+   *
+   */
+  @Event() gxMapDidLoad: EventEmitter;
+
+  @Listen("gxMapMarkerDidLoad")
+  onMapMarkerDidLoad(markerInstance) {
+    const markerElement = markerInstance.target;
+    const markerV = markerInstance.detail;
+
+    if (!!this.map) {
```

Comment:
I think there's no need for a double negation here
```suggestion
    if (this.map) {
```

Reply:
Ready.